logic/atrium-logic/src/commonMain/kotlin/ch/tutteli/atrium/logic/impl/DefaultIterableLikeAssertions.kt

```diff
@@ -250,4 +249,31 @@ class DefaultIterableLikeAssertions : IterableLikeAssertions {
             listAssertionContainer
         )
     }
+
+    override fun <T : IterableLike, E> last(
+        container: AssertionContainer<T>,
+        converter: (T) -> Iterable<E>
+    ): FeatureExtractorBuilder.ExecutionStep<T, E> =
+        collect(container, converter, "last", Iterable<E>::last)
+
+    private fun <T : IterableLike, E> collect(
+        container: AssertionContainer<T>,
+        converter: (T) -> Iterable<E>,
+        method: String,
+        collect: Iterable<E>.() -> E
+    ): FeatureExtractorBuilder.ExecutionStep<T, E> =
+        container.extractFeature
+            .methodCall(method)
+            .withRepresentationForFailure(NO_ELEMENTS)
+            .withFeatureExtraction {
+                val iterable = converter(it)
+                Option.someIf(iterable.iterator().hasNext()) {
+                    iterable.collect() ?: throw IllegalStateException(
+                        "Iterable does not hasNext() even though checked before! Concurrent access?"
+                    )
```

Comment:
this is the problem, the last element could also be `null` we don't restrict it. Hence you need to change this to:
```suggestion
                    iterable.collect()
```

Reply:
Oh, such a silly mistake. Thank you! I've pushed all the changes.